common/memory.edl

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+/*
+**==============================================================================
+**
+** memory.edl:
+**
+**     This file declares internal ECALLs/OCALLs used by liboehost/liboecore
+**     for manipulating memory allocations accross the enclave boundary.
+**
+**==============================================================================
+*/
+
+enclave
+{
+    untrusted
+    {
+        void* oe_realloc_ocall(
```

Comment:
Not needed in this PR, but we really need to think about renaming this to `oe_host_realloc_ocall()` for parity with its usage as with the rest of the host memory manipulation functions.

Reply:
Agreed. I also want to see if host malloc and host free can be implemented in EDL and put them here. Now that Anand implemented a small static buffer  for parameters I think they can.